allennlp/semparse/contexts/table_question_knowledge_graph.py

```diff
@@ -322,3 +340,47 @@ def _should_split_cell(cls, cell_text: str) -> bool:
         if ', ' in cell_text or '\n' in cell_text or '/' in cell_text:
             return True
         return False
+
+    def get_linked_agenda_items(self) -> List[str]:
+        """
+        Returns entities that can be linked to spans in the question, that should be in the agenda,
+        for training a coverage based semantic parser. This method essentially does a heuristic
+        entity linking, to provide weak supervision for a learning to search parser.
+        """
+        agenda_items: List[str] = []
+        for entity in self._get_longest_span_matching_entities():
+            agenda_items.append(entity)
+            # If the entity is a cell, we need to add the column to the agenda as well,
+            # because the answer most likely involves getting the row witht he cell.
+            if 'fb:cell' in entity:
+                agenda_items.append(self.neighbors[entity][0])
+        return agenda_items
+
+    def _get_longest_span_matching_entities(self):
```

Comment:
It'd be nice to see an explicit test of this method.

Reply:
Done.